homework1/task4.py

```diff
@@ -0,0 +1,23 @@
+"""
+Classic task, a kind of walnut for you
+Given four lists A, B, C, D of integer values,
+compute how many tuples (i, j, k, l) there are such
+that A[i] + B[j] + C[k] + D[l] is zero.
+We guarantee, that all A, B, C, D have same length of N where 0 ≤ N ≤ 1000.
+"""
+from typing import List
+
+
+def check_sum_of_four(a: List[int], b: List[int],
+                      c: List[int], d: List[int]) -> int:
+    num_of_tuples = 0
+
+    if a and b and c and d:
+        for _, i_val in enumerate(a):
+            for _, j_val in enumerate(b):
+                for _, k_val in enumerate(c):
+                    for _, l_val in enumerate(d):
```

Comment:
Complexity of this 4 for loops is O(N^4). This algorithm must be avoided.
And why do you need enumerate if you do not use it?